util/config.py

```diff
@@ -57,12 +57,18 @@ def read(self):
             # New retirement settings
             self.retirement['enabled'] = config.getboolean('Retirement', 'enabled', fallback=False)
             self.retirement['rares'] = config.getboolean('Retirement', 'Rares', fallback=True)
+            self.retirement['commons'] = config.getboolean('Retirement', 'Commons', fallback=True)
         elif 'Retirement' in config['Modules']:
             # Legacy retirement setting. Maintain classic rare retirement behavior
             self.retirement['enabled'] = config.getboolean('Modules', 'Retirement')
             self.retirement['rares'] = True
+            self.retirement['commons'] = True
         else:
             self.retirement['enabled'] = False
+            
+        if self.retirement['enabled'] and not (self.retirement['commons'] or self.retirement['rares']):
+            Logger.log_error("Invalid config. Retirement is enabled, but no ship rarities are selected.")
+            sys.exit(1)
```

Comment:
This should go inside `validate` function (around line 199), and instead of using `sys.exit(1)` you should use `self.ok = False`.